File: TransCoder_tokenized_test_set_functions/python/GENERATING_DISTINCT_SUBSEQUENCES_OF_A_GIVEN_STRING_IN_LEXICOGRAPHIC_ORDER.py
def generate ( st, s ) :
    if len ( s ) == 0 :
        return
    if s not in st :
        st . add ( s )
        for i in range ( len ( s ) ) :
            t = list ( s ) . copy ( )
            t . remove ( s [ i ] )
            t = '' . join ( t )
            generate ( st, t )
    return

File: TransCoder_tokenized_test_set_functions/java/GENERATING_DISTINCT_SUBSEQUENCES_OF_A_GIVEN_STRING_IN_LEXICOGRAPHIC_ORDER.java
static void generate ( Set < String > st, String s ) {
  if ( s . length ( ) == 0 ) {
    return;
  }
  if ( ! st . contains ( s ) ) {
    st . add ( s );
    for ( int i = 0;
    i < s . length ( );
    i ++ ) {
      String t = s;
      t = t . substring ( 0, i ) + t . substring ( i + 1 );
      generate ( st, t );
    }
  }
  return;
}